packages/@react-stately/utils/src/useControlledState.ts

```diff
@@ -43,19 +43,10 @@ export function useControlledState<T>(
 
     if (typeof value === 'function') {
       // this supports functional updates https://reactjs.org/docs/hooks-reference.html#functional-updates
-      // when someone using useControlledState calls setControlledState(myFunc)
-      // this will call our useState setState with a function as well which invokes myFunc and calls onChange with the value from myFunc
-      // if we're in an uncontrolled state, then we also return the value of myFunc which to setState looks as though it was just called with myFunc from the beginning
-      // otherwise we just return the controlled value, which won't cause a rerender because React knows to bail out when the value is the same
-      let updateFunction = (oldValue, ...functionArgs) => {
-        let interceptedValue = value(isControlled ? stateRef.current : oldValue, ...functionArgs);
-        onChangeCaller(interceptedValue, ...args);
-        if (!isControlled) {
-          return interceptedValue;
-        }
-        return oldValue;
-      };
-      setStateValue(updateFunction);
+      let oldValue = stateRef.current;
```

Comment:
Remind me again, what was this change was for? I know its from https://github.com/adobe/react-spectrum/pull/1375 but what in this PR was dependent on this fix?

Reply:
I'll update the PR with some notes on that